### PR TITLE
build: clean before building

### PR DESCRIPTION
**Pull request by ajeddeloh**
Go's caching (on by default with 1.11, enforced in 1.12) does not handle
cgo at all. Update build to always clean so it will always pick up
changes to cgo code.

### CODE REVIEW

**Comment by cgwalters** [comment]
LGTM

**Comment by lucab** [comment]
This seems to break travis cross-environment. As this is something new in 1.11, I think we should update the 1.9 travis runs with 1.11 ones here.

**Comment by ajeddeloh** [comment]
Fixed the test failures. `go clean` was deleting the `.a` the travis job created to hack together building on arm.

Looks like the BB tests _and_ tailor both fell over. Both should pass, so I'll merge without them if I can get a quick re-review

**Comment by arithx** [comment]
retest this please

**Comment by arithx** [comment]
retest this please

**Comment by arithx** [comment]
retest this please

**Comment by ajeddeloh** [comment]
retest this please

File: .travis.yml [hunk 1]
```diff
@@ -5,8 +5,8 @@ language: go
 go_import_path: github.com/coreos/ignition
 
 go:
-  - 1.8
-  - 1.9
+  - "1.10.4"
```
Comment (lucab) [inline .travis.yml]: As I don't know what we'll be using to build ignition on RHCOS and FCOS, I'd suggest to stick here to a reasonable minimum version (1.8.x seemed fine) and only use the other entries to track latest stable Go. (Unless there is some specific breakage on 1.8/1.9 and we are requiring >= 1.10 now).

Comment (ajeddeloh) [reply .travis.yml]: SGTM, fixed.

**Comment by ajeddeloh** [comment]
This has been the most painful "simple bump". Updated again to use go 1.8 and 1.11. It's got a little shell magic to not run go clean on go 1.8 since it doesn't have -cache or -testcache. Checking for that was easier than doing version comparisons in bash.

File: .travis.yml [hunk 2]
```diff
@@ -5,8 +5,8 @@ language: go
 go_import_path: github.com/coreos/ignition
 
 go:
-  - 1.8
-  - 1.9
+  - "1.8"
+  - "1.11"
```
Comment (arithx) [inline .travis.yml]: Is it worth continuing to test the two latest go versions in addition to an older version (i.e. `1.8`, `1.10`, & `1.11` in this case)?

Comment (ajeddeloh) [reply .travis.yml]: Eh, sure. travis is "free".

**Pull request merged**